### DISCUSSION TITLE
npm error: npm ERR! cb() never called!

### DISCUSSION THREAD

**Discussion by tkrah** [npm/feedback#740] [General]
Hi, I am using:

node 14.17.6 with npm 6.14.15 (I am not yet able to migrate to node 16/18 with a newer npm at the moment) and on CI/CD often this strikes us when running "npm install":

```
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community/>

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins/workspace/task-XXXX/client/build/node_cache/_logs/2022-07-28T14_25_52_989Z-debug.log
```

Looking through search results people suggest cleaning the cache or verify should help - but that does not help in this case because we have an complete new empty cache directory each run using the "--cache" option and putting the cache to the build directory (like you can see in the path of the debug log) - so there is nothing to verify or clean because the cache does not even exist before running install.

The log file at the end has this:

```
...
32080 timing npm Completed in 435553ms
32081 error cb() never called!
32082 error This is an error with npm itself. Please report this error at:
32083 error <https://npm.community>

```

So whats wrong with npm here - it does not always happen but eventually it will hit us, sometimes even 2-3 times in a row.
How to debug that further? Ideas? Suggestions?

PS: npm suggests to report this to "https://npm.community" which fails with ssl errors (but if you ignore them *yeah I know ...*) it does forward you to "https://www.npmjs.com/" and there under support you only have "Giving Feedback" which suggests opening a discussion here - if that is the wrong destination - just tell me where to report that bug instead and maybe fix / update the "please report this error ..." message of npm.

**Comment by MylesBorins** [comment]
Have you attempted to update to npm 8?

You can do this without updating to a newer version of Node.js, even in CI, with `npm install -g npm`

**Comment by Premkumarwebdeveloper** [comment]
how to fixed this error

**Reply by tkrah** [reply]
I never found a fix on that version reported. It finally disappeared when switched to node 18 and npm 8/9.
